### ISSUE TITLE
Cannot give elim a type without discriminant

### ISSUE THREAD

**Issue by wilbowma** [wilbowma/cur#6]
In the `actual-inductives` branch, `elim` cannot be given a type without some discriminant. This is stupid. See how Coq generates an eliminator for 

```
Inductive L : Type -> Type :=
| nil : (forall A, L A)
| cons : (forall A, forall a : A, L A)
```

**Comment by wilbowma** [comment]
Fixed by bluephoenix47/cur@2f5d293f9465c9bd8fa2173620dcaa9f35a7b930, but this causes an API breakage and will require more changes before it can be merged.

**Comment by wilbowma** [comment]
Actually, that changes fixes typing of elim, but breaks reduction of elim and breaks type safety. More work needed.
